modules/db/src/main/java/org/mule/module/db/internal/domain/connection/SimpleConnectionFactory.java

```diff
@@ -8,22 +8,48 @@
 package org.mule.module.db.internal.domain.connection;
 
 import java.sql.Connection;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Map;
 
 import javax.sql.DataSource;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /**
  * Creates a {@link Connection} from a {@link DataSource}
  */
 public class SimpleConnectionFactory extends AbstractConnectionFactory
 {
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private Map<String, Class<?>> typeMapping;
+
+    public SimpleConnectionFactory(Map<String, Class<?>> typeMapping)
+    {
+        this.typeMapping = typeMapping;
+    }
+
     @Override
     protected Connection doCreateConnection(DataSource dataSource)
     {
         Connection connection;
         try
         {
             connection = dataSource.getConnection();
+
+            if (typeMapping != null && !typeMapping.isEmpty())
+            {
+                try
+                {
+                    connection.setTypeMap(typeMapping);
+                }
+                catch (SQLFeatureNotSupportedException e)
+                {
+                    logger.warn("DataSource does not support custom type mappings - " + dataSource);
```

Comment:
Is there some case where we don't want to fail when this happens? If so, add it as a comment here for clarification.